codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -772,6 +772,19 @@ void fireChildReadComplete() {
             }
         }
 
+        void fireChildExceptionCaught(Throwable cause) {
+            pipeline().fireExceptionCaught(wrapStreamClosedError(cause));
+        }
+
+        private Throwable wrapStreamClosedError(Throwable cause) {
+            // If the error was caused by STREAM_CLOSED we should use a ClosedChannelException to better
+            // mimic other transports and make it easier to reason about what exceptions to expect.
+            if (cause instanceof Http2Exception && ((Http2Exception) cause).error() == Http2Error.STREAM_CLOSED) {
+                return new ClosedChannelException().initCause(cause);
```

Comment:
So how correct is this, really? I see two potential problems:
1. The channel isn't actually closed: it's only guaranteed to be half-closed and you can still receive. I don't remember how half-closure behavior works in Netty.
2. The channel isn't actually closed: there's not an impending channelInactive.

Reply:
Doh. You can semi-ignore this comment; I forgot to delete it. The other comments likely resolve the strangeness here. But the core idea of "the channel should be closed if we're triggering ClosedChannelException" remains.

Reply:
@ejona86 I think you are actual right and we should just call `channel.close()` here as well. WDYT ?

Reply:
SGTM

Reply:
Continuation of https://github.com/netty/netty/pull/7610#discussion_r163377907 ... I think we need to be careful about half closure (both for inbound and outbound events) instead of unconditionally closing.